### ISSUE TITLE
Ensure restricted access is handled gracefully

### ISSUE THREAD

**Issue by olensmar** [kubeshop/monokle#978]
Monokle currently expects cluster connections to have full access to a target cluster - need to make sure that restricted access in regard to namespaces/read/write/list/delete is handled gracefully also

**Comment by fog1985** [comment]
Hi @olensmar ,
I have created a service account inside of `monokle-rbac-testing` on testkube-dev GKE.
```
(base) taras.z.klym@lwo1-lmc-A03935:~/WORK/kubeshop/serviceaccount_creation $ kubectl --kubeconfig ./monokle-rbac-testing.config get pods
NAME    READY   STATUS    RESTARTS   AGE
nginx   1/1     Running   0          7s

(base) taras.z.klym@lwo1-lmc-A03935:~/WORK/kubeshop/serviceaccount_creation $ kubectl --kubeconfig ./monokle-rbac-testing.config delete pods nginx
Error from server (Forbidden): pods "nginx" is forbidden: User "system:serviceaccount:monokle-rbac-testing:monokle-test-serviceaccount" cannot delete resource "pods" in API group "" in the namespace "monokle-rbac-testing"
```
Will provide kube-config separately. App using this config will have these permissions:
```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  namespace: monokle-rbac-testing
  name: monokle-test-role
rules:
- apiGroups: [""] # "" indicates the core API group
  resources: ["pods", "secrets"]
  verbs: ["get", "watch", "list"]
```
If you will need more or different ones for the testing please let me know.

**Comment by fog1985** [comment]
For testing purposes there are pod and secret in this namespace:
```
(base) taras.z.klym@lwo1-lmc-A03935:~/WORK/kubeshop/serviceaccount_creation $ kubectl --kubeconfig ./monokle-rbac-testing.config get pods; kubectl --kubeconfig ./monokle-rbac-testing.config get secrets

NAME    READY   STATUS    RESTARTS   AGE
nginx   1/1     Running   0          6m44s

NAME                                      TYPE                                  DATA   AGE
db-user-pass                              Opaque                                2      18s
default-token-wqp7v                       kubernetes.io/service-account-token   3      54m
monokle-test-serviceaccount-token-jzdp4   kubernetes.io/service-account-token   3      10m
```

**Comment by olensmar** [comment]
Thanks @fog1985 - Monokle currently just fails to load any resources for this user - we'll have to fix this but I'll move it out of 1.5.0 as I think it could be a lot of work to get it right... seems we could use an approach similar to what is described at https://stackoverflow.com/questions/61724236/how-to-view-resource-permissions-with-selfsubjectrulesreview-using-curl to find out what a user has access to - and then enforce accordingly.
